elastic-search-support/src/main/java/uk/gov/hmcts/ccd/definition/store/elastic/ElasticDefinitionImportListener.java

```diff
@@ -41,42 +44,116 @@ public ElasticDefinitionImportListener(CcdElasticSearchProperties config, CaseMa
+            logMapping(caseMapping);
         } catch (Exception exc) {
             logMapping(caseMapping);
             throw new ElasticSearchInitialisationException(exc);
-        } finally {
-            if (elasticClient != null) {
-                elasticClient.close();
-            }
         }
     }
 
+    private CompletableFuture<String> handleReindexing(HighLevelCCDElasticClient elasticClient, String baseIndexName,
+                                                       String caseTypeName, String incrementedCaseTypeName,
+                                                       boolean deleteOldIndex) {
+        //initiate async elasticsearch reindexing request
+        CompletableFuture<String> reindexFuture = elasticClient.reindexData(caseTypeName, incrementedCaseTypeName)
+            .exceptionally(ex -> {
+                try {
+                    //if failed delete new index, set old index writable
+                    HighLevelCCDElasticClient asyncElasticClient = clientFactory.getObject();
+                    log.info("reindexing failed, error: {}", ex.getMessage());
+                    asyncElasticClient.removeIndex(incrementedCaseTypeName);
+                    log.info("{} deleted", incrementedCaseTypeName);
+                    asyncElasticClient.setIndexReadOnly(caseTypeName, false);
+                    log.info("{} set to writable", caseTypeName);
+                } catch (IOException e) {
+                    log.error("failed to clean up after reindexing failure", e);
+                    throw new CompletionException(e);
+                }
+                throw new CompletionException(ex);
+            });
+
+        //if success set writable and update alias to new index
+        return reindexFuture.thenApply(taskId -> {
+            try {
+                HighLevelCCDElasticClient asyncElasticClient = clientFactory.getObject();
+                log.info("updating alias from {} to {}", caseTypeName, incrementedCaseTypeName);
+                asyncElasticClient.setIndexReadOnly(baseIndexName, false);
+                asyncElasticClient.updateAlias(baseIndexName, caseTypeName, incrementedCaseTypeName);
+                if (deleteOldIndex) {
+                    log.info("deleting old index {}", caseTypeName);
+                    asyncElasticClient.removeIndex(caseTypeName);
+                }
+                return taskId;
+            } catch (IOException e) {
+                log.error("failed to clean up after reindexing success", e);
+                throw new CompletionException(e);
+            }
+        });
+    }
+
+    private String incrementIndexNumber(String indexName) {
+        String[] parts = indexName.split("-");
+        String prefix = parts[0] + "-";
+        String numberStr = parts[1];
+
+        int incremented = Integer.parseInt(numberStr) + 1;
+        @SuppressWarnings("java:S3457")
+        String formattedNumber = String.format("%0" + numberStr.length() + "d", incremented);
```

Comment:
StringUtils.leftPad(incremented,numberStr.length() , '0');
 try this method

Reply:
there will be a problem with this: what if the number incremented is a two digit number: incremented